File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/category/district/dto/DistrictDto.java
package vn.com.kns.portalapi.application.service.category.district.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.category.province.dto.ProvinceDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DistrictDto extends BaseEntityDto {
    private String name;
    private ProvinceDto province;
    private Long provinceId;
}

File: src/msa/main-service/authportal-service/src/app/shared/services/category/district/dto/district-dto.ts
import {ProvinceDto} from '../../province/dto/province-dto';

export class DistrictDto {
  id: number | null;
  name: string;
  provinceId: number | null;
  province: ProvinceDto = new ProvinceDto();

  constructor(id?: number | null, name?: string) {
    this.id = id;
    this.name = name;
  }
}
